# HG changeset patch
# User jwe
# Date 745013465 0
#      Tue Aug 10 20:11:05 1993 +0000
# Node ID 15a1c5152f3da9d22e91997f668842d87f9f8fcd
# Parent  ec4d93a11cfe3281548939b2f7a12acf077264c2
[project @ 1993-08-10 20:11:05 by jwe]
(builtin_balance): New function.

diff --git a/src/g-builtins.cc b/src/g-builtins.cc
--- a/src/g-builtins.cc
+++ b/src/g-builtins.cc
@@ -154,16 +154,34 @@ builtin_any (tree_constant *args, int na
 	  retval = new tree_constant [2];
 	  retval[0] = args[1].any ();
 	}
     }
   return retval;
 }
 
 /*
+ * Balancing for eigenvalue problems
+ */
+tree_constant *
+builtin_balance (tree_constant *args, int nargin, int nargout)
+{
+  tree_constant *retval = NULL_TREE_CONST;
+  if (nargin <= 1 || nargin > 4 || nargout < 1 || nargout > 4)
+    usage ("[aa {,dd}] = balance (a, {opt}) or \n\
+[aa, bb {,cc, dd}] = balance (a, b {,opt}), opt = 'P' or 'S'");
+  else
+    {
+      DLD_BUILTIN (args, nargin, nargout, balance,
+		   retval = balance (args, nargin, nargout));
+    }
+  return retval;
+}
+
+/*
  * Clear the screen?
  */
 tree_constant *
 builtin_clc (tree_constant *args, int nargin, int nargout)
 {
   rl_beg_of_line ();
   rl_kill_line (1);
   rl_clear_screen ();
